cmd/kubeadm/app/phases/addons/proxy/manifests.go

```diff
@@ -77,6 +77,10 @@ spec:
       - name: kube-proxy
         image: {{ .Image }}
         imagePullPolicy: IfNotPresent
+        resources:
+          requests:
+            cpu: 100m
+            memory: 50Mi
```

Comment:
50Mi is fine for small clusters, but definitely way too small for larger ones (or to be more specific for clusters with big enough number of services and their backends)

Reply:
@wojtek-t do you have an upper bound?

Reply:
In 5k-node clusters with 150k pods, each pod being a backend for some service I've seen things like 300MB IIRC.
But we probably also don't want to allocate that much in every cluster.